test/TransitivePathTest.cpp

```diff
@@ -701,6 +701,112 @@ TEST_P(TransitivePathTest, zeroLength) {
+
+    EXPECT_TRUE(T->isBoundOrId());
+
+    auto resultTable = T->computeResultOnlyForTesting(requestLaziness());
+    assertResultMatchesIdTable(resultTable, expected);
+  }
+
+  {
+    TransitivePathSide left(std::nullopt, 0, 1337, 0);
+    TransitivePathSide right(std::nullopt, 1, Variable{"?target"}, 1);
+    auto T = makePathUnbound(
+        std::move(sub), {Variable{"?start"}, Variable{"?target"}}, left, right,
+        0, std::numeric_limits<size_t>::max(), std::move(index));
+
+    EXPECT_TRUE(T->isBoundOrId());
+
+    auto resultTable = T->computeResultOnlyForTesting(requestLaziness());
+    assertResultMatchesIdTable(resultTable, expected);
+  }
+
+  {
+    TransitivePathSide left(std::nullopt, 0, Variable{"?start"}, 0);
+    TransitivePathSide right(std::nullopt, 1, 1337, 1);
+    auto T = makePathUnbound(
+        std::move(sub), {Variable{"?start"}, Variable{"?target"}}, left, right,
+        0, std::numeric_limits<size_t>::max(), std::move(index));
+
+    EXPECT_TRUE(T->isBoundOrId());
+
+    auto resultTable = T->computeResultOnlyForTesting(requestLaziness());
+    assertResultMatchesIdTable(resultTable, expected);
+  }
+}
+
+// _____________________________________________________________________________
+TEST_P(TransitivePathTest, literalsNotInIndex) {
+  std::string index = "<a> a 0 , 1 , 2 , 4 .";
+
+  auto sub = makeIdTableFromVector(
+      {
+          {0, 2},
+          {2, 4},
+      },
+      Id::makeFromInt);
+
+  auto expected = IdTable{2, ad_utility::testing::makeAllocator()};
+
+  {
+    TransitivePathSide left(std::nullopt, 0, 1337, 0);
+    TransitivePathSide right(std::nullopt, 1, 1337, 1);
+    auto T = makePathUnbound(
+        sub.clone(), {Variable{"?start"}, Variable{"?target"}}, left, right, 1,
+        std::numeric_limits<size_t>::max(), index);
+
+    EXPECT_TRUE(T->isBoundOrId());
+
+    auto resultTable = T->computeResultOnlyForTesting(requestLaziness());
+    assertResultMatchesIdTable(resultTable, expected);
+  }
+
+  {
+    TransitivePathSide left(std::nullopt, 0, 1337, 0);
+    TransitivePathSide right(std::nullopt, 1, Variable{"?target"}, 1);
+    auto T = makePathUnbound(
+        std::move(sub), {Variable{"?start"}, Variable{"?target"}}, left, right,
+        1, std::numeric_limits<size_t>::max(), std::move(index));
+
+    EXPECT_TRUE(T->isBoundOrId());
+
+    auto resultTable = T->computeResultOnlyForTesting(requestLaziness());
+    assertResultMatchesIdTable(resultTable, expected);
+  }
+
+  {
+    TransitivePathSide left(std::nullopt, 0, Variable{"?start"}, 0);
+    TransitivePathSide right(std::nullopt, 1, 1337, 1);
+    auto T = makePathUnbound(
+        std::move(sub), {Variable{"?start"}, Variable{"?target"}}, left, right,
+        1, std::numeric_limits<size_t>::max(), std::move(index));
+
+    EXPECT_TRUE(T->isBoundOrId());
+
+    auto resultTable = T->computeResultOnlyForTesting(requestLaziness());
+    assertResultMatchesIdTable(resultTable, expected);
+  }
+}
+
```

Comment:
The most important feature imho in this PR is not yet tested, namely a fixed constant that is a local vocab index that was added e.g. by an update, so 

`<kartoffelsalat> <p>+ ?y` 
Where the input for <p> contains `<kartoffelsalat>`  but only as a local vocab index (for the test create it manually, but in reality this might happen in case of updates.